From 8d1b28391de13a390637b2203fe4aca16c71fdb7 Mon Sep 17 00:00:00 2001
From: Jens Wiklander <jens.wiklander@linaro.org>
Date: Fri, 1 Dec 2023 11:54:17 +0100
Subject: [PATCH 0086/1695] core: add interrupt_get_main_chip_may_fail()

Add interrupt_get_main_chip_may_fail() to be able to check if it's
possible to use interrupts without causing a panic.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 621cdbd19e617a0bb3de9b1de99de59f7cd9cbf2 core: add interrupt_get_main_chip_may_fail()]
Change-Id: I5a2474b48410eb5ab4a97bb235465ab8a2bbbaea
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>
[etienne: cherry-picked from 245a552c4164e6803517ea796a1deacfa8259574]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/include/kernel/interrupt.h | 2 ++
 core/kernel/interrupt.c         | 5 +++++
 2 files changed, 7 insertions(+)

diff --git a/core/include/kernel/interrupt.h b/core/include/kernel/interrupt.h
index 370f4330f..22a7a0449 100644
--- a/core/include/kernel/interrupt.h
+++ b/core/include/kernel/interrupt.h
@@ -150,6 +150,8 @@ void interrupt_main_init(struct itr_chip *data);
 
 /* Retrieve main interrupt controller reference */
 struct itr_chip *interrupt_get_main_chip(void);
+/* Retrieve main interrupt controller reference, or NULL on failure */
+struct itr_chip *interrupt_get_main_chip_may_fail(void);
 
 #ifdef CFG_DT
 /*
diff --git a/core/kernel/interrupt.c b/core/kernel/interrupt.c
index f46a7700b..4fe92bd55 100644
--- a/core/kernel/interrupt.c
+++ b/core/kernel/interrupt.c
@@ -46,6 +46,11 @@ struct itr_chip *interrupt_get_main_chip(void)
 	return itr_main_chip;
 }
 
+struct itr_chip *interrupt_get_main_chip_may_fail(void)
+{
+	return itr_main_chip;
+}
+
 #ifdef CFG_DT
 int dt_get_irq_type_prio(const void *fdt, int node, uint32_t *type,
 			 uint32_t *prio)
-- 
2.47.2

